### ISSUE TITLE
The requested record was not found or you do not have sufficient permissions to view it.

### ISSUE THREAD

**Issue by gaarabyakuya** [DynCon365/SharePointDragAndDropPCF#4]
I keep getting this error "The requested record was not found or you do not have sufficient permissions to view it." I already re do the app several times, I change the SP guid. thanks in advance

**Comment by anupamkhasia** [comment]
@gaarabyakuya - you need to place "Sharepoint site record Id" from CRM not sharepoint site GUID . I have same error and resolved it by placing sharepoint site record Id. 

Let me know if it works.

**Comment by gaarabyakuya** [comment]
@anupamkhasia Thank you, that indeed solve that error. Could tell me how did you map the fields? because the control keeps saying loading and nothing happens. I tried any combination that I can think of but no luck.

**Comment by anupamkhasia** [comment]
@gaarabyakuya - I am also still trying to load control as it looks there are additional sharepoint permissions needs to be given .

**Comment by ThoshanTharanga** [comment]
@anupamkhasia  have you find any solution for file loading issue? For me also it keeps loading nothing happen...

**Comment by anupamkhasia** [comment]
@ThoshanTharanga - Yes , we have to configure sharepoint site and than you will be able to use this component. Let me know if you are facing any other errors.

**Comment by ThoshanTharanga** [comment]
@anupamkhasia  Thanks for the reply.... Could I know what kind of configuration needs to be done there..... seems it is giving error when retrieving the sharepoint folder on onFileDropped funtion.
![image](https://user-images.githubusercontent.com/79072814/123535056-f1e5c200-d764-11eb-9a6c-f8d0438a1110.png)

SharePoint setup mean, is it kind of setting up permission..? Thanks :)

**Comment by anupamkhasia** [comment]
@ThoshanTharanga - I mean to have dynamics crm and sharepoint integration configuration we need to do. please refer to following link 

https://www.crmsoftwareblog.com/2020/02/how-to-setup-a-dynamics-365-integration-with-sharepoint/

Let me know if you have done this. 

Thanks
Anupam

**Comment by ThoshanTharanga** [comment]
@anupamkhasia Yeah that integration part is done and selected the based on entity as Account. Still it keep loading... and can see the console error which I have mentioned previously. 
Thanks,
Thoshan

**Comment by anupamkhasia** [comment]
@ThoshanTharanga - can you give me screenshot of PCF control configuration you have setup ?

**Comment by ThoshanTharanga** [comment]
@anupamkhasia Sure... for the client id I have given the client id of the app which I have registered in Azure App Directory and for the site id I have given the "sharepointsiteid" field from api/data/v9.0/sharepointsites response. 
![image](https://user-images.githubusercontent.com/79072814/123583710-6f700780-d823-11eb-932f-9e300f7b5f6e.png)
![image](https://user-images.githubusercontent.com/79072814/123583791-90385d00-d823-11eb-88c5-726c3fb9bd36.png)

**Comment by anupamkhasia** [comment]
@ThoshanTharanga - not sure and can not determine how to look into your error but here is my configuration . please try this and let know. 
![image](https://user-images.githubusercontent.com/61499146/123632982-3ab97b80-d836-11eb-908b-e209291db547.png)

**Comment by kshitijkhanna** [comment]
Hello @ThoshanTharanga @anupamkhasia : I am facing same issue. We're you able to get this working?

**Comment by ThoshanTharanga** [comment]
HI @kshitijkhanna . Yes, this error is coming because sometimes the input we passing as a parameter in getSharePointFolderName() funtion is undefined in SharePointUploadControlIVM.ts file. Could you try by adding below change to it?
![image](https://user-images.githubusercontent.com/79072814/139367142-5d95d8c3-5363-4a1c-aefe-c24e5a5c292f.png)

if this fix work for you, I will create a PR to that change.

Thank you :)

**Comment by kshitijkhanna** [comment]
Hi @ThoshanTharanga :- Thanks ! The code change works and we are able to upload documents however its does not pick the Title before the folder name and creates folder with just ID( **__2DA820A3E83BEC118C620012476F46B0_** ). 

We get the below **console error for GET.**

https://xyz.sharepoint.com/sites/abc/_api/web/getFolderByServerRelativePath(decodedUrl='abcd%2F_2DA820A3E83BEC118C620012476F46B0') 

For a normal documents control, it does pickup Title before the ID and creates it properly.

**Comment by kdot088** [comment]
> HI @kshitijkhanna . Yes, this error is coming because sometimes the input we passing as a parameter in getSharePointFolderName() funtion is undefined in SharePointUploadControlIVM.ts file. Could you try by adding below change to it? ![image](https://user-images.githubusercontent.com/79072814/139367142-5d95d8c3-5363-4a1c-aefe-c24e5a5c292f.png)
> 
> if this fix work for you, I will create a PR to that change.
> 
> Thank you :)

Hi, I am also having the same issue and tried this code. unfortunately it doesn't work. do you have any idea why? I am still getting the same 'replace' error.

**Comment by kdot088** [comment]
<img width="463" alt="zz" src="https://user-images.githubusercontent.com/45902106/227426603-0ce79d0f-f5a7-4fd4-a2bf-2864a73cb014.PNG">

**Comment by NataliWinOn365** [comment]
Did anybody solved all the errors? :)

**Comment by Maza95** [comment]
Hello, please can any one tell me how i can import the solution i downloaded the repository and then i run
npm install and then npm fix after that pac pcf push command but it didnt work. I also tried to build the project butit gave me the same results.
can anyone help me please with the steps after downloading the repo.

thanks a lot
Best regards
